@kroening This still isn't compiling on cbmc yet: 
```
value_set_fi.cpp:858:48: error: within this context
           index_expr.make_typecast(array.type());
```

So will hold off on the bump for now.

--- a/src/ansi-c/expr2c.cpp
+++ b/src/ansi-c/expr2c.cpp
@@ -1939,9 +1939,11 @@ std::string expr2ct::convert_constant(
       if(src.operands().size()!=1)
         return convert_norep(src, precedence);
 
-      if(src.op0().id()==ID_constant)
+      const auto &annotation = to_unary_expr(src).op();
+
+      if(annotation.id() == ID_constant)
       {
-        const irep_idt &op_value=src.op0().get(ID_value);
+        const irep_idt &op_value = to_constant_expr(annotation).get_value();
 
         if(op_value=="INVALID" ||
            has_prefix(id2string(op_value), "INVALID-") ||
@@ -1951,7 +1953,7 @@ std::string expr2ct::convert_constant(
           return convert_norep(src, precedence);
       }
       else
-        return convert_with_precedence(src.op0(), precedence);
+        return convert_with_precedence(annotation, precedence);
     }
   }
   else if(type.id()==ID_string)

--- a/src/goto-programs/remove_function_pointers.cpp
+++ b/src/goto-programs/remove_function_pointers.cpp
@@ -391,13 +391,13 @@ void remove_function_pointerst::remove_function_pointer(
     t3->make_goto(t_final, true_exprt());
 
     // goto to call
-    address_of_exprt address_of(fun, pointer_type(fun.type()));
+    const address_of_exprt address_of(fun, pointer_type(fun.type()));
 
-    if(address_of.type()!=pointer.type())
-      address_of.make_typecast(pointer.type());
+    const auto casted_address =
+      typecast_exprt::conditional_cast(address_of, pointer.type());
 
     goto_programt::targett t4=new_code_gotos.add_instruction();
-    t4->make_goto(t1, equal_exprt(pointer, address_of));
+    t4->make_goto(t1, equal_exprt(pointer, casted_address));
   }
 
   // fall-through

--- a/src/pointer-analysis/value_set_fi.cpp
+++ b/src/pointer-analysis/value_set_fi.cpp
@@ -853,9 +853,13 @@ void value_set_fit::get_reference_set_sharing_rec(
         index_exprt index_expr(
           object, from_integer(0, index_type()), expr.type());
 
+        exprt casted_index;
+
         // adjust type?
         if(object.type().id() != "#REF#" && object.type() != array_type)
-          index_expr.make_typecast(array.type());
+          casted_index = typecast_exprt(index_expr, array.type());
+        else
+          casted_index = index_expr;
 
         offsett o = a_it->second;
         mp_integer i;
@@ -868,7 +872,7 @@ void value_set_fit::get_reference_set_sharing_rec(
         else
           o.reset();
 
-        insert(dest, index_expr, o);
+        insert(dest, casted_index, o);
       }
     }
 

--- a/src/pointer-analysis/value_set_fivr.cpp
+++ b/src/pointer-analysis/value_set_fivr.cpp
@@ -964,9 +964,13 @@ void value_set_fivrt::get_reference_set_sharing_rec(
         index_exprt index_expr(
           object, from_integer(0, index_type()), expr.type());
 
+        exprt casted_index;
+
         // adjust type?
         if(object.type().id() != "#REF#" && object.type() != array_type)
-          index_expr.make_typecast(array.type());
+          casted_index = typecast_exprt(index_expr, array.type());
+        else
+          casted_index = index_expr;
 
         offsett o = a_it->second;
         mp_integer i;
@@ -979,7 +983,7 @@ void value_set_fivrt::get_reference_set_sharing_rec(
         else
           o.reset();
 
-        insert_from(dest, index_expr, o);
+        insert_from(dest, casted_index, o);
       }
     }
 

--- a/src/pointer-analysis/value_set_fivrns.cpp
+++ b/src/pointer-analysis/value_set_fivrns.cpp
@@ -657,9 +657,13 @@ void value_set_fivrnst::get_reference_set_rec(
         index_exprt index_expr(
           object, from_integer(0, index_type()), expr.type());
 
+        exprt casted_index;
+
         // adjust type?
-        if(object.type() != array_type)
-          index_expr.make_typecast(array.type());
+        if(object.type().id() != "#REF#" && object.type() != array_type)
+          casted_index = typecast_exprt(index_expr, array.type());
+        else
+          casted_index = index_expr;
 
         offsett o = a_it->second;
         mp_integer i;
@@ -672,7 +676,7 @@ void value_set_fivrnst::get_reference_set_rec(
         else
           o.reset();
 
-        insert_from(dest, index_expr, o);
+        insert_from(dest, casted_index, o);
       }
     }
 

--- a/src/util/format_expr.cpp
+++ b/src/util/format_expr.cpp
@@ -104,9 +104,7 @@ static std::ostream &format_rec(std::ostream &os, const multi_ary_exprt &src)
 
   std::string operator_str = id2string(src.id()); // default
 
-  if(
-    src.id() == ID_equal && !src.operands().empty() &&
-    src.op0().type().id() == ID_bool)
+  if(src.id() == ID_equal && to_equal_expr(src).op0().type().id() == ID_bool)
   {
     operator_str = u8"\u21d4"; // <=>, U+21D4
   }

--- a/src/util/simplify_expr.cpp
+++ b/src/util/simplify_expr.cpp
@@ -1123,8 +1123,9 @@ bool simplify_exprt::simplify_if(if_exprt &expr)
       else if(falsevalue.is_true())
       {
         // a?b:1 <-> !a OR b
-        or_exprt tmp(boolean_negate(cond), truevalue);
-        simplify_node(tmp.op0());
+        exprt tmp_not_cond = boolean_negate(cond);
+        simplify_node(tmp_not_cond);
+        or_exprt tmp(tmp_not_cond, truevalue);
         simplify_node(tmp);
         expr.swap(tmp);
         return false;
@@ -1140,8 +1141,9 @@ bool simplify_exprt::simplify_if(if_exprt &expr)
       else if(truevalue.is_false())
       {
         // a?0:b <-> !a && b
-        and_exprt tmp(boolean_negate(cond), falsevalue);
-        simplify_node(tmp.op0());
+        exprt tmp_not_cond = boolean_negate(cond);
+        simplify_node(tmp_not_cond);
+        and_exprt tmp(tmp_not_cond, falsevalue);
         simplify_node(tmp);
         expr.swap(tmp);
         return false;

